packages/react-native/Libraries/AppDelegate/RCTRootViewFactory.mm

```diff
@@ -0,0 +1,286 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#import "RCTRootViewFactory.h"
+#import "RCTAppDelegate.h"
+#import <React/RCTCxxBridgeDelegate.h>
+#import <React/RCTLog.h>
+#import <React/RCTRootView.h>
+#import <React/RCTSurfacePresenterBridgeAdapter.h>
+#import <React/RCTUtils.h>
+#import <react/renderer/runtimescheduler/RuntimeScheduler.h>
+#import "RCTAppSetupUtils.h"
+
+#if RN_DISABLE_OSS_PLUGIN_HEADER
+#import <RCTTurboModulePlugin/RCTTurboModulePlugin.h>
+#else
+#import <React/CoreModulesPlugins.h>
+#endif
+#import <React/RCTBundleURLProvider.h>
+#import <React/RCTComponentViewFactory.h>
+#import <React/RCTComponentViewProtocol.h>
+#import <React/RCTFabricSurface.h>
+#import <React/RCTSurfaceHostingProxyRootView.h>
+#import <React/RCTSurfacePresenter.h>
+#import <ReactCommon/RCTContextContainerHandling.h>
+#if USE_HERMES
+#import <ReactCommon/RCTHermesInstance.h>
+#else
+#import <ReactCommon/RCTJscInstance.h>
+#endif
+#import <ReactCommon/RCTHost+Internal.h>
+#import <ReactCommon/RCTHost.h>
+#import <ReactCommon/RCTTurboModuleManager.h>
+#import <react/config/ReactNativeConfig.h>
+#import <react/renderer/runtimescheduler/RuntimeScheduler.h>
+#import <react/renderer/runtimescheduler/RuntimeSchedulerCallInvoker.h>
+#import <react/runtime/JSRuntimeFactory.h>
+
+static NSString *const kRNConcurrentRoot = @"concurrentRoot";
+
+static NSDictionary *updateInitialProps(NSDictionary *initialProps, BOOL isFabricEnabled)
+{
+  NSMutableDictionary *mutableProps = [initialProps mutableCopy] ?: [NSMutableDictionary new];
+  // Hardcoding the Concurrent Root as it it not recommended to
+  // have the concurrentRoot turned off when Fabric is enabled.
+  mutableProps[kRNConcurrentRoot] = @(isFabricEnabled);
+  return mutableProps;
+}
+
+@implementation RCTRootViewFactoryConfiguration
+
+- (instancetype)initWithBundleURL:(NSURL *)bundleURL {
+  
+  if (self = [super init]) {
+    self.bundleURL = bundleURL;
+    // Set default properties
+    self.fabricEnabled = [self newArchEnabled];
+    self.turboModuleEnabled = [self newArchEnabled];
+    self.bridgelessEnabled = false;
+  }
+  return self;
+}
+
+- (BOOL)newArchEnabled
+{
+#if RCT_NEW_ARCH_ENABLED
+  return YES;
+#else
+  return NO;
+#endif
+}
```

Comment:
one nice thing of having this at the RCTAppDelegate level is that the AppDelegate subclass in the user app can override the method and decide programmatically whether the new architecture is enabled or not.
Allow to install cocoapods only once, with the flag turned on, and control the architecture from code without the need of reinstalling the pods.

I think we should restore this possibility. By reimplementing the method here, we are not honouring the value set in the `AppDelegate` and in the `RCTAppDelegate` anymore.
We should probably pass the value to the factory, rather then reimplementing the method.

Reply:
Also, we can use directly `RCTIsNewArchEnabled()` as we do down below, so this method can completely go.

Reply:
Actually, it's properly honored. It's used to just retrieve the defaults when using `RCTRootViewFactory` without `RCTAppDelegate`. But when used together `RCTAppDelegate` overrides this setting: 

```objc
configuration.fabricEnabled = self.fabricEnabled;
configuration.turboModuleEnabled = self.turboModuleEnabled;
configuration.bridgelessEnabled = self.bridgelessEnabled;
```
 
I thought about using `RCTIsNewArchEnabled()` but it has the same issue, `RCTAppDelegate` is responsible for setting it.. So those two classes will be dependent on each other to work properly.

Reply:
But if we are changing the initializer for `RCTRootViewFactoryConfiguration` settings those defaults don't make sense. I've removed it 👍🏻